### DISCUSSION TITLE
How do you use copilot to deploy an application (scheduled job) to a new region?

### DISCUSSION THREAD

**Discussion by john-walton** [aws/copilot-cli#5401] [Q&A] [answered]
copilot version: v1.31.0

I have successfully deployed to a test environment on region us-east-1 using my profile profile-1.

[profile-1]
region = us-east-1

Now, trying to deploy to region eu-west-1 using profile-2.

[profile-2]
region = eu-west-1

This says it provisions successfully, but I am unable to deploy.

Running as profile-1:
export AWS_PROFILE=profile-1
copilot env init --name test --app my-app --profile profile-2 --region eu-west-1
✔ Manifest file for environment test already exists at copilot/environments/test/manifest.yml, skipping writing it.
- Update regional resources with stack set "my-app-infrastructure"  [succeeded]  [10.1s]
- Update regional resources with stack set "my-app-infrastructure"  [succeeded]        [125.6s]
  - Update resources in region "eu-west-1"                                      [create complete]  [123.4s]
    - KMS key to encrypt pipeline artifacts between stages                      [create complete]  [123.4s]
    - ECR container image repository for "my-app-job"               [create complete]  [0.0s]
    - S3 Bucket to store local artifacts                                        [create complete]  [1.9s]
✔ Proposing infrastructure changes for the my-app-test environment.
- Creating the infrastructure for the my-app-test environment.  [create complete]  [45.9s]
  - An IAM Role for AWS CloudFormation to manage resources                  [create complete]  [18.0s]
  - An IAM Role to describe resources in your environment                   [create complete]  [18.8s]
✔ Provisioned bootstrap resources for environment test in region eu-west-1 under application my-app.
Recommended follow-up actions:
  - Update your manifest copilot/environments/test/manifest.yml to change the defaults.
  - Run `copilot env deploy --name test` to deploy your environment.

Trying to deploy:
copilot env deploy --name test
✘ upload artifacts for environment test: upload custom resources to bucket XXX: upload custom resource "CertificateValidationFunction": upload manual/scripts/custom-resources/certificatevalidationfunction/xxx.zip to bucket XXX: AccessDenied: Access Denied
        status code: 403, request id: XXX, host id: XXX

What am I missing?

**Comment by KollaAdithya** [comment]
Hello @john-walton !
If you want to create an `test` environment in region `eu-west-1` with `profile-2`.
I think you are initializing the environment with ` profile-2` and trying to deploy the environment with `profile-1`.
Can you try the following steps.
```
1. export AWS_PROFILE=profile-2
2. export AWS_REGION=eu-west-1
3. copilot env init --name test --app my-app --profile profile-2 --region eu-west-1
4. copilot env deploy --name test
```

**Reply by john-walton** [reply]
In that case I get the following error on the 'copilot env init':

copilot env init --name test --app my-app --profile profile-2 --region eu-west-1
✘ get application my-app configuration: couldn't find an application named my-app in account XXX and region eu-west-1

**Reply by KollaAdithya** [reply]
@john-walton
In which region and which profile have you used to create your application ?

**Reply by john-walton** [reply]
Initially I used profile-1 in us-east-1. Everything deployed and the job can be started there.
The problem is trying to do the same in another region (using same or different profile).

Let me try a fresh start with a new app and profiles. 

I can see from my first try above where I got the access denied error there are some artifacts getting created (I see them in cloud formation).

I'll post back after some more trials.

**Comment by KollaAdithya** [comment, marked as answer]
Sorry @john-walton for all the churn.

When deploying the environment you need to use the same profile that is used for creating the app.
```
export AWS_PROFILE=app-profile # profile used for the app.
copilot env init --name test --app my-app --profile profile-2
copilot env deploy
```

**Reply by john-walton** [reply]
Thanks, sorry for long wait on response. Ran into permission issues with new profile.
